src/physics/em/detail/UrbanMscScatter.hh

```diff
@@ -163,48 +177,43 @@ UrbanMscScatter::UrbanMscScatter(const UrbanMscRef&       shared,
 template<class Engine>
 CELER_FUNCTION auto UrbanMscScatter::operator()(Engine& rng) -> MscInteraction
 {
-    MscInteraction result{true_path_, inc_direction_, {0, 0, 0}};
-
-    // Do not sample scattering at the last or at a small step
-    if (true_path_ < helper_.range() && true_path_ > geom_min())
+    if (skip_sampling_)
     {
-        auto end_energy = helper_.end_energy(true_path_);
+        // Do not sample scattering at the last or at a small step
+        return {true_path_, inc_direction_, {0, 0, 0}};
```

Comment:
return MscInteraction{true_path_, inc_direction_, {0, 0, 0}};  (to be consistent with the return at the bottom?).